### ISSUE TITLE
Build owncloud-android Project

### ISSUE THREAD

**Issue by Bruno-Piccinin-RealComm** [owncloud/android#800]
Hi,

I have a problem when i try to build owncloud-android Project.
All problems start from **NotificationBuilderWithProgressBar.java** with:

*  **@Override
    public NotificationCompat.Builder setProgress(int max, int progress, boolean indeterminate) {
        mContentView.setProgressBar(R.id.progress, max, progress, indeterminate);
        if (max > 0) {
            mContentView.setViewVisibility(R.id.progressHolder, View.VISIBLE);
        } else {
            mContentView.setViewVisibility(R.id.progressHolder, View.GONE);
        }
        return this;
    }**

*  **@Override
    public Notification build() {
        Notification result = super.build();
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB) {
            // super.build() in Android 2.x totally ruins whatever was made #setContent 
            result.contentView = mContentView;
        }
        return result;
    }**

Eclipse wants to remove the **"@Override" annotation**.

So I did it and I edited this two methods like this:

* **//@Override  REMOVED Override ANNOTATION
    public NotificationCompat.Builder setProgress(int max, int progress, boolean indeterminate) {
        mContentView.setProgressBar(R.id.progress, max, progress, indeterminate);
        if (max > 0) {
            mContentView.setViewVisibility(R.id.progressHolder, View.VISIBLE);
        } else {
            mContentView.setViewVisibility(R.id.progressHolder, View.GONE);
        }
        return this;
    }**

* **//@Override REMOVED Override ANNOTATION
    public Notification build() {
        //Notification result = super.build(); REMOVED AND CHANGED result INITIALIZATION
        Notification result = new Notification();
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB) {
            // super.build() in Android 2.x totally ruins whatever was made #setContent 
            result.contentView = mContentView;
        }
        return result;
    }**

and then i removed all referer to these methods in:

* **FileSyncAdapter.java**
* **FileUploader.java**
* **FileDownloader.java**

commenting all:
* **.setProgress(...)**
* **mNotificationManager.notify(...)**
* **mNotificationBuilder.setProgress(...)**

After that, Eclipse managed to compile and create the APK of owncloud-android Project, but when the application starts, it's crash with message *"Unfortunately, owncloud has stopped."*

The Logcat tracks this issue:

`FATAL EXCEPTION: main
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{com.owncloud.android/com.owncloud.android.ui.activity.FileDisplayActivity}: java.lang.ClassNotFoundException: Didn't find class "com.owncloud.android.ui.activity.FileDisplayActivity" on path: /data/app/com.owncloud.android-2.apk
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2106)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2230)
android.app.ActivityThread.access$600(ActivityThread.java:141)
android.app.ActivityThread$H.handleMessage(ActivityThread.java:1234)
android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:137)
at android.app.ActivityThread.main(ActivityThread.java:5041)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:511)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560)
at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.owncloud.android.ui.activity.FileDisplayActivity" on path: /data/app/com.owncloud.android-2.apk
dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:65)
at java.lang.ClassLoader.loadClass(ClassLoader.java:501)
at java.lang.ClassLoader.loadClass(ClassLoader.java:461)
at android.app.Instrumentation.newActivity(Instrumentation.java:1054)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2097)
... 11 more`

How can I solve all these?

Thanks
Best Regards

**Comment by tobiasKaminsky** [comment]
Please have a look on #829 
Maybe both of you can figure out what the problem is.

**Comment by weichen55** [comment]
I have the same problem as Bruno.  I built the apk file using maven. 
when i installed  it on Android pad 4.4.2, it crashed and the log says displayfileactivity class is not found.
Is this problem still open? Thanks.

**Comment by Bruno-Piccinin-RealComm** [comment]
Hi,

I just wanted to tell you that I finally compiled the project! But to do that, I had to use ANDROID STUDIO and not ECLIPSE.

So if anyone as a problems like I had, try to compile project using ANDROID STUDIO. The only thing that i had to do was to import the project, change the minimum skd Version in file .Manifest from 8 to 16 version, and comment some catch code because ANDROID STUDIO reported that codes as a error because it would never be executed.
